spring-integration-event/src/main/java/org/springframework/integration/event/inbound/ApplicationEventListeningMessageProducer.java

```diff
@@ -116,20 +129,45 @@ public void onApplicationEvent(ApplicationEvent event) {
 		}
 	}
 
+	@Override
+	protected Object evaluatePayloadExpression(Object rootObject) {
+		if (!this.payloadExpressionSet && rootObject instanceof PayloadApplicationEvent) {
+			return ((PayloadApplicationEvent) rootObject).getPayload();
```

Comment:
Thanks, @snicoll this whole discussion is moot - I thought Artem had added the ability to evaluate an expression against the event when it's been there since day 1. So I now agree, that it's correct to unwrap the payload.

I had mis-interpreted the javadoc..

     * <p>Mainly intended for internal use within the framework.

to mean that publishing arbitrary objects was for internal use whereas it really means the wrapper is.